omod-common/src/main/java/org/openmrs/module/webservices/rest/web/resource/impl/BasePageableResult.java

```diff
@@ -58,6 +67,11 @@ public SimpleObject toSimpleObject(Converter preferredConverter) throws Response
 				links.add(context.getPreviousLink());
 			ret.add("links", links);
 		}
+		if (context.getRequest() != null) {

```

Comment:
yes getParameter fails if if request in context is null. ok, changed RequestContext and added tests for it to RequestContextTest.